### PR TITLE
add delete_by_field method in RedisVectorStore

### PR DESCRIPTION
**Pull request by minglu7**
# Description

The addition of the delete_by_field method provides more flexibility and functionality to the RedisVectorStore class compared to the previous delete method.  The delete method allows for deletion of nodes based on a specific document ID (ref_doc_id). This means that to delete a node, you must know its exact document ID. This can be limiting in situations where you want to delete nodes based on other criteria.  On the other hand, the delete_by_field method allows developers to delete nodes based on a specific field name and field value. This means you can delete nodes based on any field, not just the document ID. This is particularly useful when dealing with large amounts of data and needing to delete data based on specific conditions

### CODE REVIEW

**Comment by logan-markewich** [comment]
@minglu7 we recently added delete_nodes and get_nodes to the base vector store class.

You can get/delete by node ids or filters.

Let's implement these methods instead? So far been slowly implementing these in vector stores as it comes up

(`clear()` was also added, to delete everything in the index)

**Comment by logan-markewich** [comment]
Reference
https://github.com/run-llama/llama_index/blob/267ae33537cbda88093a97d93786bbd6034566f7/llama-index-core/llama_index/core/vector_stores/types.py#L323

File: llama-index-integrations/vector_stores/llama-index-vector-stores-redis/llama_index/vector_stores/redis/base.py [hunk 1]
```diff
@@ -266,7 +266,7 @@ def add(self, nodes: List[BaseNode], **add_kwargs: Any) -> List[str]:
             key.strip(self._index.prefix + self._index.key_separator) for key in keys
         ]
 
-    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-redis/llama_index/vector_stores/redis/base.py:269]: Delete is method on the base class, we can't remove it

Comment (minglu7) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-redis/llama_index/vector_stores/redis/base.py:269]: i see ,  thx

**Pull request closed without merging**